Added tests for uncovered files and removed unused imports

watchdogoblivion opened the pull request:
Removed unused imports and dependencies across all packages. Added tests for JSONEntities and exceptions.

afcabezasq committed:
coverage

watchdogoblivion committed:
Removed additional Junit depenedencies from pom.xml since they are not used. Added tests for JSONEntities and exceptions. Added test for main simply for full coverage. Added tests to complete Module testing. Added new tests to testNGsuite.xml

afcabezasq committed:
util tests

watchdogoblivion committed:
Merge pull request #203 from revaturelabs/Features_Test_sdorilas

For class InvalidRequestExceptionTest, the method InvalidModuleException_MessageInjectedBySuper_TestTrue should be called
InvalidRequestExceptionTest _MessageInjectedBySuper_TestTrue. Will fix in next commit. Performing merge.

watchdogoblivion committed:
Removed unused imports across all packages. Changed aforementioned method name in class InvalidRequestExceptionTest. Removed the test MetricsControllerTest from the testing suite, since it is stated to be broken and needs to be replaced. Removed previously added Jacoco dependency since it will not be used. Replaced all Junit assert imports with TestNG assert imports since TestNG is our testing suite.

howardlt94 committed:
Merge pull request #204 from revaturelabs/Features_Test_sdorilas

Removed all unused dependencies, imports, and Annotations

afcabezasq committed:
ValidationUtil 98% coverage

afcabezasq committed:
ControllerLogAspect 100% coverage

afcabezasq commented:
It is ok

howardlt94 reviewed (dismissed):
daamn looks really good

watchdogoblivion committed:
Added full coverage for Metrics Controller. Added Equals method to TimeGraphData.java since it was missing and causing incorrect result in MetricsControllerTest. Added test for TimeGraphData equals and hash method. Add a null check in MetricsController since it was missing and leading to null pointer exceptions when trying to test conditions.

watchdogoblivion committed:
Added full coverage for LinkControllerTest. Added MetricsControllerTest to testNGsuite.xml since it was missing.

watchdogoblivion committed:
Added full coverage for ContentControllerTest

afcabezasq committed:
Removed redundant code

watchdogoblivion committed:
Merge branch 'Features_Test_SAHF' into Features_Test_Andres

watchdogoblivion committed:
Merge pull request #209 from revaturelabs/Features_Test_Andres

Looks good, Merge approved.

watchdogoblivion committed:
Corrections from previous merge. Removed no longer existing test from suite, Fixed incorrect package name in suite. Commented out failing test form ExceptionAspect. Tests are now working properly.

howardlt94 committed:
Merge branch 'Features_Test_SAHF' into Features_Test_controllers_sdorilas

howardlt94 committed:
Merge pull request #207 from revaturelabs/Features_Test_controllers_sdorilas

Features test controllers sdorilas

afcabezasq committed:
acomplished 100% Coverage for UtilTest

watchdogoblivion committed:
Resolved merge from Features_Test_Andres into Features_Test_SAHF

watchdogoblivion committed:
Renamed some packages to fallow naming conventions. Removed unused imports. Update suite with correct package names

carolynrehm commented on src/main/java/com/revature/cmsforce/CMSforceApplication.java:
```diff
@@ -25,8 +25,11 @@
 
 }
+
+interface Test{
```
What is this interface for?

watchdogoblivion replied:
I think that was a simple overlook. We will remove it.

carolynrehm commented on src/main/java/com/revature/util/ExceptionAspect.java:
```diff
@@ -11,18 +11,15 @@
-@Autowired
-Logging log;
+	@Autowired
+	public Logging log;
```
What was the reason for this change - Does this need to be public?

watchdogoblivion replied:
No, it does not need to be public. This was a temporary change that Andres made. As I mentioned before, the project was not following best practices in terms of test package naming. So He was not able to access the Logging object. I told him to rename the test package so he can access the class, so the field can be default package-private. He simply forgot to change the access modifier.

carolynrehm replied:
Got it, can we remove this if it's not necessary ?

carolynrehm replied:
And update the test package names?

carolynrehm commented on src/test/java/com/revature/cmsforce/CMSforceApplicationTest.java:
```diff
@@ -0,0 +1,14 @@
+
+	// Test class added ONLY to cover main() invocation not covered by application tests.
+	@Test
+	public void mainTest() {
```
Is this a meaningful test? Test coverage is not meaningful if it's not actually testing anything

watchdogoblivion replied:
I contemplated that repeatedly, it was a choice that I made. It is not a very meaningful test hence the comment above the method itself. I was researching the importance of full coverage and when would it be b=necessary to test the main method. It seems it is only for aesthetic purposes and differs from company to company. I can remove it if you would like.,

carolynrehm replied:
Gotcha. I understand the motivation but I would remove it

carolynrehm commented on src/main/java/com/revature/util/ControllerLogAspect.java:
```diff
@@ -10,6 +10,8 @@
 @Aspect
 public class ControllerLogAspect { //Logging for the controllers
 	
+	
```
Try to only include files in pull requests that are relevant to the issue

carolynrehm commented on src/test/java/com/revature/controllertests/ContentControllerTest.java:
```diff
@@ -30,99 +27,147 @@
-	
+
+	@Test
+	public void givenValidDataCreateContent_GetLinksNotNull() throws Exception {
```
method naming conventions; shouldn't include an underscore

watchdogoblivion replied:
Okay, I will change that. I saw a few conventions so I simply chose that one. I will go to strictly camelcase.

carolynrehm replied:
relevant in areas like src/test/java/com/revature/entities/ModuleTest.java as well

carolynrehm commented on src/test/java/com/revature/entities/ContentTest.java:
```diff
@@ -131,7 +130,7 @@ public void testGetLinks() {
 	@Test
 	public void testSetLinks() {
-		Set<Link> link = new HashSet<Link>();
+//		Set<Link> link = new HashSet<Link>();
```
remove > commenting out - does this test need to be here?

watchdogoblivion replied:
I commented it out because I was not sure if I was allowed to remove old content at the time. I will remove it right now.

carolynrehm replied:
Feel free to remove things if they're not needed 👍

carolynrehm commented on src/test/java/com/revature/entities/ModuleTest.java:
```diff
@@ -0,0 +1,282 @@
+	
+	
+	
+	//Any time that two nulls appear in a test of a constructor, that is for a feature that was created after the tests were created to allow them to pass.
```
remove commented out code

watchdogoblivion committed:
Removed unused interface Test. Renamed any method using underscores with camelcase. Changed access modifier for Logging ExceptionAspect.Removed Main method test. Removed unsed method from ContentTest.Removed commented out code from ModuleTest

watchdogoblivion commented:
@carolynrehm I made all the changes and tested that the suite still worked. Please let me know if there is anything else required.

carolynrehm commented on src/main/java/com/revature/controllers/MetricsController.java:
```diff
@@ -41,64 +40,62 @@
 	@PostMapping("/{timeFrame}")
-	public MetricsData getMetrics(@PathVariable("timeFrame") long timeRange, 
-								  @RequestBody Map<String, Object> filter) {		
+	public MetricsData getMetrics(@PathVariable("timeFrame") long timeRange, @RequestBody Map<String, Object> filter) {
```
This method needs refactoring.
- why is it using a post request
- why are the filter parameters being passed into requestbody rather than as request parameters? map directly into an array or list instead of using StringTokenizer
- logic should be refactored (e.g. lines 62-69 have contradictory conditions that result in the same thing) and moved to a service method

carolynrehm replied:
This doesn't need to be fixed in this pull request, as it's not really in scope. I'm not sure why there need be changes to this file for this pull request though. Perhaps handle this elsewhere.

watchdogoblivion replied:
Hi @carolynrehm , most of the changes here are from IDE formatting. I changed three things. First, the StringTokenizer initialization because it was missing a null check which leads to an unreachable branch for testing at line 62. Second I removed the unchecked suppression since it was not necessary. Third, I removed the null check for idsIn since it will always get initialized at line 48. What I failed to realize was that the branches in line 62 were exactly the same. So I will remove the unnecessary if/else.

carolynrehm replied:
Gotcha. Definitely some fixes that we can make but can we keep it out of this pull request just because it's not related to testing?

watchdogoblivion replied:
Good to know, I will try not to modify too many files in the future.

carolynrehm commented on src/test/java/com/revature/controllertests/MetricsControllerTest.java:
```diff
@@ -1,270 +1,182 @@
+import com.revature.util.TimeGraphData;
 
+@SpringBootTest(classes = CMSforceApplication.class)
+public class MetricsControllerTest extends AbstractTestNGSpringContextTests {
```
Looks like there are a lot of removed tests here, was that intentional? Why were they removed?

watchdogoblivion replied:
Yes. I replaced them per the comment on the class below:
* This test class fails to test the functionality of the controller,
 * not to mention the controller is nearly untestable at this point.
 * These tests should be deprecated and replaced when the controller
 * is fixed.

carolynrehm replied:
got it, thanks for the clarification!

carolynrehm commented on src/test/java/com/revature/controllertests/RequestControllerTest.java:
```diff
@@ -88,7 +88,7 @@ public void givenValidDataCreateRequest() throws Exception {
 				.content(objMapper.writeValueAsString(rc.requestToJSONRequest(request))));
 
-		JSONRequest ret = objMapper.readValue(result.andReturn().getResponse()
+		objMapper.readValue(result.andReturn().getResponse()
```
is this even needed in this test at all?

watchdogoblivion replied:
I did not make that change, but it looks like it is not needed I will go change it.

carolynrehm commented on src/test/java/com/revature/util/ValidationUtilTest.java:
```diff
@@ -0,0 +1,153 @@
+		content.setDescription(badString);
+		vu.verifyContent(content);
+	}
+	@Test
```
what is this test doing?

carolynrehm replied:
a few other tests in this class as well

watchdogoblivion replied:
@afcabezasq could you please shed some light here, since these were assigned to you.

carolynrehm reviewed (changes requested):
Why even have src/main/java/com/revature/util/Logging.java? Is it being used?

Also the testing of beans is fine here, but in the future it's not necessary. Code coverage should be maximized for services and controllers, but generated code does not need to be tested.

Also, non javadoc comments are not good practice to have throughout your code. (JSONContentTest, JSONModuleTest, etc) 

As mentioned in other comments, src/main/java/com/revature/controllers/MetricsController.java doesn't seem to need to be changed for this pull request but should definitely be refactored in the future.

There should be one thing tested per test method; if a test fails, there would be one reason it fails.
When a controller test expects a status and uses an assert method, there is more than one reason. This is also present in the ExceptionAspectTest and ControllerAspectTest

watchdogoblivion committed:
Applied Carolyns change request. Removing comments and separatiing methods for controllers. Module Controller and ContentController complete

watchdogoblivion committed:
Removed redundant verify call

watchdogoblivion committed:
Cleaned COntentCOntrollerTest

watchdogoblivion committed:
Cleaned MetricsCOntrollerTest nd RequestCOntrollerTest

watchdogoblivion committed:
Finished ControllerTest cleanup

watchdogoblivion committed:
Renamed last two packages to match proper convention. Relfected insuite.xml. Removed additional comments. Fixed metod naming convention incertain classes

watchdogoblivion committed:
All of the fchanges where made to all tests except the Service test. Fred will address it. Removed all unecessary comments and tests from al ltest files. Removed the Logging class since it is pointless. Split tests to test only one specific thing.

watchdogoblivion commented:
**_Why even have src/main/java/com/revature/util/Logging.java? Is it being used?_**
* Logging.java has been removed

**_Also the testing of beans is fine here, but in the future, it's not necessary. Code coverage should be maximized for services and controllers, but generated code does not need to be tested._**
* I agree. I told everyone that when starting the project, but it was pointed out that the entities were being tested, so we wanted to hold up consistency with what was already there.

**_Also, non-javadoc comments are not good practice to have throughout your code. (JSONContentTest, JSONModuleTest, etc)_**
* Removed all non-javadoc comments from every test, and added java doc comments where necessary

**_As mentioned in other comments, src/main/java/com/revature/controllers/MetricsController.java doesn't seem to need to be changed for this pull request but should definitely be refactored in the future._**
* If we have time outside of writing tests for the front end, we will designate someone to refactor it.

**_There should be one thing tested per test method; if a test fails, there would be one reason it fails.
When a controller test expects a status and uses an assert method, there is more than one reason. This is also present in the ExceptionAspectTest and ControllerAspectTest_**
* Separated assertions per method as requested.

carolynrehm commented:
Updates look good. The only thing left unresolved is ValidationUtilTest. Once we get some more details there, this should be ready to merge @watchdogoblivion @afcabezasq

afcabezasq commented on src/test/java/com/revature/util/ValidationUtilTest.java:
```diff
@@ -0,0 +1,153 @@
+	public void verifyContent_TestFifthConditionThrowException() {
+		content.setDescription(badString);
+		vu.verifyContent(content);
+	}
```
Basically, theese tests verify that the exception is trhown when there is an  Empty String or when a string has more  than 254 characters

carolynrehm replied:
There still seem to be methods which neither have an assert method nor check for an exception. Please refactor these as they should not be included. Also, refactor the method signatures to following naming conventions. Thanks!

We can merge when this is complete

afcabezasq committed:
removed useless test

watchdogoblivion committed:
Merge pull request #216 from revaturelabs/Features_Test_Andres

Approved

afcabezasq committed:
refactors

watchdogoblivion committed:
Merge pull request #220 from revaturelabs/Features_Test_Andres

Well done

carolynrehm commented on src/test/java/com/revature/util/ValidationUtilTest.java:
```diff
@@ -41,102 +41,102 @@ void init() {
 	}
 	@Test
-	public void verifyFilter_Test() {
+	public void verifyFilterTest() {
```
@afcabezasq what are these last three testing for?

afcabezasq replied:
They are testing one if branch on  verifyId(),verifyListModuleId() and verifyFilter()  respectively

watchdogoblivion replied:
@afcabezasq Please reply here.

carolynrehm reviewed (changes requested):
There are still multiple assertions in JSONModuleTest(line ~25) JSONRequestTest (~line 28). Update these and pull from dev. Then ping me and I'll merge these in!

watchdogoblivion committed:
Separated multiple asserts from method

watchdogoblivion committed:
Merge branch 'Features_Test_SAHF' of https://github.com/revaturelabs/cms-content-service into Features_Test_SAHF

watchdogoblivion committed:
Resolved conflicts, refactored entity tests for easy implementation, reoved unused imports

watchdogoblivion committed:
Resolved merge. Fixed some tests and coverage.

watchdogoblivion commented:
@carolynrehm , I pulled all changes from dev and made additions and changes where I saw necessary.
